opt-dist: make llvm builds optional

ognevny opened the pull request:
adds command line option for disabling llvm builds. it's useful in case of user having their own optimized LLVM, so they won't waste time for (at least) 3 LLVM builds. in this case PGO optimized rustc will be already built in Stage 1, so rust-lang/rust#143898 should be addressed for this change

couldn't test locally on Linux laptop due to small SSD storage, will try now with windows-msvc host

<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r? <reviewer name>
-->
<!-- homu-ignore:end -->
r? @Kobzol 

try-job: dist-x86_64-linux
try-job: dist-x86_64-msvc

ognevny commented:
> couldn't test locally on Linux laptop due to small SSD storage, will try now with windows-msvc host

after finally managing to make a good environment, I tested this and the change works as intended

Kobzol commented on src/tools/opt-dist/src/exec.rs:
```diff
@@ -132,10 +132,13 @@ impl Bootstrap {
     }
 
     pub fn llvm_pgo_instrument(mut self, profile_dir: &Utf8Path) -> Self {
```
Shouldn't you be modifying `llvm_pgo_optimize` instead? Also, please use `Option<LlvmPGOProfile>` instead of passing the profile an empty string.

Kobzol commented on src/tools/opt-dist/src/main.rs:
```diff
@@ -255,32 +263,37 @@ fn execute_pipeline(
+    };
 
-    let bolt_profiles = if env.use_bolt() {
+    let bolt_profiles = if env.build_llvm() && env.use_bolt() {
```
This disables BOLT for rustc if you don't build LLVM. I don't suppose many people will need that exact scenario, but we should still support it, otherwise `build_llvm` is a misleading name.

Kobzol reviewed (commented):
Left two nits, will start a try run in the meantime.

Kobzol commented:
@bors2 try

ognevny committed:
opt-dist: make llvm builds optional

adds command line option for disabling llvm builds. it's useful in case of user having their own
optimized LLVM, so they won't waste time for (at least) 3 LLVM builds. in this case PGO optimized
will be already built in Stage 1, so my previous PR should be addressed for this change

rust-timer commented:
Queued d08b35280fb8e6cb4ea877781c9531e0e69b9278 with parent 1aa5b2246560ce85b42fa8b33e5927c5de3fa389, future [comparison URL](https://perf.rust-lang.org/compare.html?start=1aa5b2246560ce85b42fa8b33e5927c5de3fa389&end=d08b35280fb8e6cb4ea877781c9531e0e69b9278).
There are currently 0 preceding artifacts in the [queue](https://perf.rust-lang.org/status.html).
It will probably take at least ~1.1 hours until the benchmark run finishes.
<!-- rust-timer: temporary -->

Kobzol commented:
@bors r+ rollup
